ql/src/java/org/apache/hadoop/hive/ql/io/AcidDirectory.java

```diff
@@ -0,0 +1,230 @@
+    this.oldestBase = oldestBase;
+  }
+
+  public void setUnCompactedAborts(boolean unCompactedAborts) {
+    this.unCompactedAborts = unCompactedAborts;
+  }
+
+  /**
+   * Is Base directory in raw format or in Acid format
+   */
+  public boolean isBaseInRawFormat() {
+    return base != null && base.isRawFormat();
+  }
+
+  /**
+   * Get the list of original files.  Not {@code null}.  Must be sorted.
+   * @return the list of original files (eg. 000000_0)
+   */
+  public List<HadoopShims.HdfsFileStatusWithId> getOriginalFiles() {
+    return originalFiles;
+  }
+
+  /**
+   * List of original directories containing files in not ACID format
+   */
+  public List<Path> getOriginalDirectories() {
+    return originalDirectories;
+  }
+
+  /**
+   * Get the list of delta directories that are valid and not
+   * obsolete.  Not {@code null}.  List must be sorted in a specific way.
+   * See {@link org.apache.hadoop.hive.ql.io.AcidUtils.ParsedDeltaLight#compareTo(org.apache.hadoop.hive.ql.io.AcidUtils.ParsedDeltaLight)}
+   * for details.
+   * @return the minimal list of current directories
+   */
+  public List<AcidUtils.ParsedDelta> getCurrentDirectories() {
+    return currentDirectories;
+  }
+
+  /**
+   * Get the list of obsolete directories. After filtering out bases and
+   * deltas that are not selected by the valid transaction/write ids list, return the
+   * list of original files, bases, and deltas that have been replaced by
+   * more up to date ones.  Not {@code null}.
+   */
+  public List<Path> getObsolete() {
+    return obsolete;
+  }
+
+  /**
+   * Get the list of directories that has nothing but aborted transactions.
+   * @return the list of aborted directories
+   */
+  public List<Path> getAbortedDirectories() {
+    return abortedDirectories;
+  }
+
+  /**
+   * Get the list of writeIds that belong to the aborted transactions.
+   * @return the list of aborted writeIds
+   */
+  public Set<Long> getAbortedWriteIds() {
+    return abortedWriteIds;
+  }
+
+  /**
+   * Does the directory contain writeIds that belong to aborted transactions,
+   * but are mixed together with committed writes. These aborted writes can not be cleaned.
+   * @return true if there are aborted writes that can can be cleaned
+   */
+  public boolean hasUncompactedAborts() {
+    return unCompactedAborts;
+  }
+
+  public FileSystem getFs() {
+    return fs;
+  }
+
+  /**
+   * Delete deltas that should be read by this reader.
+   */
+  public List<AcidUtils.ParsedDelta> getDeleteDeltas() {
+    return currentDirectories.stream().filter(AcidUtils.ParsedDeltaLight::isDeleteDelta).collect(Collectors.toList());
+  }
+
+  /**
+   * All original, base and delta bucket files that should be read by this reader
+   * @throws IOException ex
+   */
+  public List<AcidUtils.AcidBaseFileInfo> getBaseAndDeltaFiles() throws IOException {
+    List<AcidUtils.AcidBaseFileInfo> baseAndDeltaFiles = new ArrayList<>();
+    if (base == null) {
```

Comment:
If it is a normal acid table and contains only a few delta, this branch will do nothing, because there can not be original files in the table (the delta files will be added later from the currentDirectories). This path is here (if I understood it correctly) for tables that were converted to transactional, but not yet compacted.